vendor/github.com/OpenBazaar/spvwallet/txstore.go

```diff
@@ -354,14 +354,15 @@ func (ts *TxStore) Ingest(tx *wire.MsgTx, height int32) (uint32, error) {
 		}
 		// Let's check the height before committing so we don't allow rogue peers to send us a lose
 		// tx that resets our height to zero.
-		if txn.Height <= 0 {
-			ts.Txns().UpdateHeight(tx.TxHash(), int(height))
+		if err == nil && txn.Height <= 0 {
```

Comment:
I would've added an `else` to the prior block and nest the `Height` check inside so it's obvious the logic can only go one way or the other to the next reader instead of being surprised this separate block of logic is conditioned on the same data as the prior block.